### PR TITLE
[New] add `sort-default-props` and deprecate `jsx-sort-default-props`

### PR DESCRIPTION
**Pull request by alexzherdev**
Pt. 1 of #1834

### CODE REVIEW

**Comment by alexzherdev** [comment]
Looks like I need to fix #1817 to improve coverage

File: lib/rules/jsx-sort-default-props.js [hunk 1]
```diff
@@ -130,12 +134,6 @@ module.exports = {
             checkSorted(propTypesObject.properties);
           }
           break;
-        case 'CallExpression':
-          const innerNode = node.arguments && node.arguments[0];
```
Comment (ljharb) [inline lib/rules/jsx-sort-default-props.js]: hm, why is this unused exactly?

Could this change be done in a separate commit/PR?

Comment (alexzherdev) [reply lib/rules/jsx-sort-default-props.js]: See discussion in https://github.com/yannickcr/eslint-plugin-react/issues/1817#issuecomment-396329150
I removed it because coveralls was red and I needed to increase coverage to make it pass 🙂Otherwise I can absolutely make this change separately

Comment (ljharb) [reply lib/rules/jsx-sort-default-props.js]: it'd be great to at least make it separate commits, even if it stays in this PR :-)

Comment (alexzherdev) [reply lib/rules/jsx-sort-default-props.js]: @ljharb looks like this change isn't necessary for coverage now, so I'll submit a separate PR when this one is in.

**Comment by alexzherdev** [comment]
Just rebased to resolve conflicts

**Comment by VincentLanglet** [comment]
If my PR is merged before yours, you'll have to add my bug fix.
https://github.com/yannickcr/eslint-plugin-react/pull/2182

**Comment by VincentLanglet** [comment]
@alexzherdev Since my PR https://github.com/yannickcr/eslint-plugin-react/pull/2182 was merged, you have to update the new sort-default-props, or you'll lose the fix I made.

Thanks ;)

**Comment by amannn** [comment]
What's the status of this? Looks like this PR was approved and the only thing left to do is to copy over the bugfix mentioned above.

**Comment by ljharb** [comment]
ping @alexzherdev

**Comment by alexzherdev** [comment]
Done

File: lib/rules/sort-default-props.js [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+      MemberExpression(node) {
+        if (!isDefaultPropsDeclaration(node)) {
+          return;
+        }
+
+        checkNode(node.parent.right);
+      }
+    };
+  }
+};
```
Comment (ljharb) [inline lib/rules/sort-default-props.js]: needs a trailing newline

Comment (alexzherdev) [reply lib/rules/sort-default-props.js]: Ah, was on a different laptop 👍

**Comment by ljharb** [comment]
This looks good, but I'm going to cut a patch release before merging semver-minor items.
